<?xml version="1.0"?>
<launch>
  <!-- allow custom models to be stored in this ros workspace -->
  <env name="GAZEBO_MODEL_PATH" value="$(find plating_demo)/models"/>
  <!-- startup simulated world -->
  <node name="gazebo" pkg="gazebo_ros" type="gzserver" respawn="false" output="screen"
	args="-e ode $(find plating_demo)/worlds/ground_picture.world" />
  <!-- start gazebo client
  <node name="gazebo_gui" pkg="gazebo_ros" type="gzclient" respawn="false" output="screen"/> -->

  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find plating_demo)/urdf/ur5e_robot.urdf.xacro'" />
 
  <!-- push robot_description to factory and spawn robot in gazebo -->
  <node name="spawn_gazebo_model" pkg="gazebo_ros" type="spawn_model" args="-urdf -param robot_description -model robot -z 0.0" respawn="false" output="screen" />

  <include file="$(find ur_gazebo)/launch/controller_utils.launch"/>

  <rosparam file="$(find ur_gazebo)/controller/arm_controller_ur5.yaml" command="load"/>
  <node name="arm_controller_spawner" pkg="controller_manager" type="controller_manager" args="spawn arm_controller" respawn="false" output="screen"/>

</launch>
